index.bs

```diff
@@ -488,13 +490,238 @@ partial interface NeuralNetworkContext {
+                  optional Operand bias, optional Operand recurrentBias,
+                  optional boolean resetAfter = true,
+                  optional RecurrentNetworkWeightLayout layout = "zrn",
+                  optional sequence<RecurrentNetworkActivation> activations);
+};
+</script>
+<div algorithm=grucell>
+    **Arguments:**
+        - *hiddenSize*: a {{long}} scalar. The value of the second dimension of the output tensor shape. It indicates the number of features in the hidden state.
+        - *input*: an {{Operand}}. The input 2-D tensor of shape [batch_size, input_size]. 
+        - *weight*: an {{Operand}}. The 2-D input weight tensor of shape [3 * hidden_size, input_size]. The ordering of the weight vectors in the first dimension of the tensor shape is specified according to the *layout* argument.
+        - *recurrentWeight*: an {{Operand}}. The 2-D recurrent weight tensor of shape [3 * hidden_size, hidden_size]. The ordering of the weight vectors in the first dimension of the tensor shape is specified according to the *layout* argument.
+        - *hidden*: an {{Operand}}. The 2-D input hidden state tensor of shape [batch_size, hidden_size].
+        - *bias*: an optional {{Operand}}. The 1-D input bias tensor of shape [3 * hidden_size]. The ordering of the bias vectors in the first dimension of the tensor shape is specified according to the *layout* argument.
+        - *recurrentBias*: an optional {{Operand}}. The 1-D recurrent bias tensor of shape [3 * hidden_size]. The ordering of the bias vectors in the first dimension of the tensor shape is specified according to the *layout* argument.
+        - *resetAfter*: an optional {{boolean}} indicating whether to apply the reset gate after or before matrix multilication. Default to true.
+        - *layout*: an optional {{RecurrentNetworkWeightLayout}}. The ordering of the weight and bias vectors for the internal gates of GRU, specifically the *update (z)*, *reset (r)*, and *new (n)* gate, as indicated in the first dimension of the weight and bias tensor shapes. When not specified, the default layout is *"zrn"*.
+        - *activations*: an optional sequence of {{RecurrentNetworkActivation}}. A pair of activation functions with the first function used for the update and reset gate, and the second used for the new gate. When not specified, it's default to the sigmoid ("sigmoid") and the hyperbolic tangent ("tanh") function respectively.
+
+    **Returns:** an {{Operand}}. The 2-D tensor of shape [batch_size, hidden_size], the cell output of a single time step of the recurrent network.
+
+    <div class="note">
+    The behavior of this operation with default argument values can be generically emulated from the usage of other operations as follow. However, user agents typically have a more efficient implementation for it, therefore its usage is encouraged from the performance standpoint.
+    <pre highlight="js">
+    let one = nn.constant(1);
+    let zero = nn.constant(0);
+
```

Comment:
Would you like to illustrate the handling of `layout` and `activations`?

Reply:
> You mean in the informative sample code section?

Yes. 

Or at least we add some comments in the sample code that mentions this implementation is just for default `layout` and `activations` case. What do you think?

Reply:
Another reason is I am thinking about referring to the sample code for webnn-polyfill implementation. I guess it would also help other implementers.

For example the handling of `activations`,
```js
let activation1 = nn.sigmoid;
let activation2 = nn.tanh;
if (activations) {
  activation1 = nn[activations[0]];
  activation2 = nn[activations[1]];
}

let z = activation1(....);

let r = activation1(....);

let n = activation2(....);
```

Reply:
> Or at least we add some comments in the sample code that mentions this implementation is just for default `layout` and `activations` case.

I just realized that I missed the text "The behavior of this operation with default argument values". Apologize for that. It looks good to me.